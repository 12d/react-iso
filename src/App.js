/**
 * @author xuweichen@meitu.io
 * @date 2017/8/24
 */
import React,{Component} from 'react';
// import PropTypes from 'prop-types';
// import ReactDOM from 'react-dom';
// import css from './css/base.css';
import {
    Route,
    Link
} from 'react-router-dom'
import Home from './Home';
import List from './List';
import Detail from './pages/Detail';
import AsyncComponent from './common/AsyncComponent';
export default class App extends Component {
    render(){
        return (
                <div>
                    <Route exact path="/" component={Home}/>
                    <Route exact path="/list" component={List}/>
                    <Route path="/detail/:id" render={AsyncComponent(Detail)}/>
                </div>
        )
    }
    constructor(){
        super()
    }
    componentDidMount(){

    }
}