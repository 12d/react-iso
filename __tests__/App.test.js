/**
 * @author xuweichen@meitu.io
 * @date 2017/8/24
 */
global.requestAnimationFrame = function(callback) {
    setTimeout(callback, 0);
};
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';



test('renders correctly', ()=>{
    const component = renderer.create(
        <App/>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

})