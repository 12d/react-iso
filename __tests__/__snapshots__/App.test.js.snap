// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h1>
  By react-ssr
</h1>
`;
